src/Cardano/Prelude/GHC/Heap/NormalForm/Classy.hs

```diff
@@ -296,6 +311,19 @@ instance NoUnexpectedThunks (AllowThunk a) where
   noUnexpectedThunks _ _ = return NoUnexpectedThunks
   whnfNoUnexpectedThunks = noUnexpectedThunks
 
+instance (HasFields s a, FromSymbols s, Generic a, Typeable a, GWhnfNoUnexpectedThunks (Rep a))
+   => NoUnexpectedThunks (AllowThunksIn s a) where
+  showTypeOf _ = show $ typeRep (Proxy @a)
+  whnfNoUnexpectedThunks ctxt (AllowThunksIn x) = gWhnfNoUnexpectedThunks allowedThunks ctxt fp
+    where
+      fp :: Rep a x
+      !fp = from x
+      allowedThunks = fromSymbols (Proxy @s)
+
+class HasFields (s :: [Symbol]) a
```

Comment:
The only reason for this class is to make sure the user doesn't have a typo I guess? Maybe document its purpose.